src/NuGet.Clients/NuGet.SolutionRestoreManager/RestoreOperationLogger.cs

```diff
@@ -464,7 +471,7 @@ private int GetMSBuildOutputVerbositySetting()
         {
             ThreadHelper.ThrowIfNotOnUIThread();
 
-            var dte = _serviceProvider.GetDTE();
+            var dte = _taskFactory.Run(async () => await _asyncServiceProvider.GetDTEAsync());
```

Comment:
This makes additional dependency for threadpool thread to get the dte instance even though its already on UI thread which will create problems when threadpool is already exhausted. So since this is a private api make it async or if you can't then just use `GetDte`.